src/Symfony/Component/Scheduler/Tests/Generator/MessageGeneratorTest.php

```diff
@@ -128,6 +129,35 @@ public function testYieldedContext()
         $this->assertEquals(self::makeDateTime('22:16:00'), $context->nextTriggerAt);
     }
 
+    public function testCheckpointSavedInBrokenLoop()
+    {
+        $now = self::makeDateTime('22:12:00');
+
+        $clock = $this->createMock(ClockInterface::class);
```

Comment:
You can use ClockMock here

Reply:
I did that to keep consistency with other tests, but yeah it's would be better